### DISCUSSION TITLE
Check user auth in functions?

### DISCUSSION THREAD

**Discussion by Godefroy** [nhost/nhost#1019] [Tips and Tricks]
I don't find any example in the doc or in the repo, only super basic express routes.
Do you have a more complete example and some good practices?

I guess I can use the Authorization header and use it somehow.
Maybe there is a middleware for that?

In Firebase we get a context with auth: https://firebase.google.com/docs/reference/functions/firebase-functions.https.callablecontext.md

**Comment by plmercereau** [comment]
Hello again Godefroy,

Yes, you can use the `authorization` header, assuming the client sent it in their request (which is automatically done when using `nhost.functions.call`).

In your function, you can then decode the JWT from the received headers - here's an example of how it would look, inspired from [the `getUserClaims` method of our Stripe library](https://github.com/nhost/nhost/blob/main/packages/stripe-graphql-js/src/utils.ts):
```js
import jwt from 'jsonwebtoken'

export default (req, res) => {
  const authorizationHeader = req.headers.get('authorization')
  const accessToken = authorizationHeader?.split(' ')[1]
  if (!accessToken) {
    return res.status(401).send('no jwt')
  }
  const jwtSecret = JSON.parse(process.env.NHOST_JWT_SECRET)
  const decodedToken = jwt.verify(accessToken, jwtSecret.key)
  const hasuraClaims = decodedToken['https://hasura.io/jwt/claims']
  if (!hasuraClaims?.['x-hasura-user-id']) {
    return res.status(401)
  }
  res.status(200).send(`Hiya`)
}

**Comment by elitan** [comment]
Also see [Get the authenticated user in a Serverless Function](https://github.com/nhost/nhost/discussions/278).

**Comment by Godefroy** [comment]
Thanks, it's working 😃 

It would be useful to have some helpers provided and a little more documentation. Do you need some help?

**Reply by elitan** [reply]
Help is always welcomed. Either with PRs directly or if you want to discuss more in detail feel free to reach out to me: johan@nhost.io
